### DISCUSSION TITLE
[Feature]: DataTable Search Optimization

### DISCUSSION THREAD

**Discussion by archiewood** [evidence-dev/evidence#2430] [General]
### Description

Improve speed of returning results when searching in DataTable

### What problem would this solve?

Currently, when searching for longer strings in the datatable component, the query takes longer to execute if you type in the query character by character vs if you paste in the search query.

For reproduction, try this [storybook](https://645d58d2e049d64800e0ee15-crzyxbwzxh.chromatic.com/?path=/story/viz-datatable--with-search-limit-100000) 

![CleanShot 2024-07-23 at 10 53 16](https://github.com/user-attachments/assets/715303e9-9033-4374-ad9c-058c2169cd8f)

**Comment by ItsMeBrianD** [comment]
https://github.com/evidence-dev/evidence/blob/next/packages/lib/sdk/src/usql/query/Query.js#L1214-L1224

Likely going to involve some tuning here, along with debouncing in the datatable to ensure we aren't creating Queries not needed
